consensus/state_sync/helpers.go

```diff
@@ -6,12 +6,12 @@ import (
 	"google.golang.org/protobuf/types/known/anypb"
 )
 
-func (m *stateSync) SendStateSyncMessage(stateSyncMsg *typesCons.StateSyncMessage, peerId cryptoPocket.Address, blockHeight uint64) error {
+func (m *stateSync) SendStateSyncMessage(stateSyncMsg *typesCons.StateSyncMessage, msgType string, peerId cryptoPocket.Address, blockHeight uint64) error {
 	anyMsg, err := anypb.New(stateSyncMsg)
 	if err != nil {
 		return err
 	}
-	m.logger.Info().Uint64("height", blockHeight).Msg(typesCons.SendingStateSyncMessage(peerId, blockHeight))
+	m.logger.Info().Uint64("height", blockHeight).Msg(typesCons.SendingStateSyncMessage(peerId, msgType, blockHeight))
```

Comment:
In `types/message.go`, we use `ProtoReflect().Descriptor().Name()`

```go
func getMessageType(msg Message) string {
	return string(msg.ProtoReflect().Descriptor().Name())
}
```

I think you might be able to use this and remove the msgType string altogether.

Reply:
Using this way returns StateSync, and not the type of the oneOf field, i.e. GetBlockRequest etc. Therefore I implemented getMessageType function as:

```go
	switch msg.Message.(type) {
	case *typesCons.StateSyncMessage_MetadataReq:
		return "MetadataRequest"
	case ...
	}
```